core/components/atoms/input/__tests__/Input.test.tsx

```diff
@@ -209,3 +209,773 @@ describe('Input component selection range is set correctly on input type change'
+      expect(wrapper.style.minWidth).toBe('256px');
+    });
+
+    it('does not apply default minWidth for number input type', () => {
+      const { getByTestId } = render(<Input name="test" type="number" />);
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      expect(wrapper.style.minWidth).toBe('');
+    });
+
+    it('applies custom minWidth when provided', () => {
+      const { getByTestId } = render(<Input name="test" type="number" minWidth="400px" />);
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      expect(wrapper.style.minWidth).toBe('400px');
+    });
+  });
+
+  describe('Input Element Size-Specific CSS Classes', () => {
+    it('applies tiny-specific input styling classes', () => {
+      const { getByTestId } = render(<Input name="test" size="tiny" value="test" />);
+
+      const input = getByTestId('DesignSystem-Input');
+      expect(input).toHaveClass('Input-input--tiny');
+    });
+
+    it('applies regular-specific input styling classes', () => {
+      const { getByTestId } = render(<Input name="test" size="regular" value="test" />);
+
+      const input = getByTestId('DesignSystem-Input');
+      expect(input).toHaveClass('Input-input--regular');
+    });
+
+    it('applies large-specific input styling classes', () => {
+      const { getByTestId } = render(<Input name="test" size="large" value="test" />);
+
+      const input = getByTestId('DesignSystem-Input');
+      expect(input).toHaveClass('Input-input--large');
+    });
+  });
+
+  describe('Input Element Utility Class Behaviors', () => {
+    it('applies margin-right utility class when actionIcon is present', () => {
+      const customActionIcon = <Input.ActionButton name="visibility" onClick={jest.fn()} />;
+      const { getByTestId } = render(<Input name="test" value="test value" actionIcon={customActionIcon} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      expect(input).toHaveClass('mr-4');
+    });
+
+    it('applies margin-right utility class when onClear is provided', () => {
+      const onClearMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" value="test value" onClear={onClearMock} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      expect(input).toHaveClass('mr-4');
+    });
+
+    it('applies margin-right utility class when info tooltip is provided', () => {
+      const { getByTestId } = render(<Input name="test" value="test value" info="This is helpful information" />);
+
+      const input = getByTestId('DesignSystem-Input');
+      expect(input).toHaveClass('mr-4');
+    });
+
+    it('does not apply margin-right utility class when no right-side elements are present', () => {
+      const { getByTestId } = render(<Input name="test" value="test value" />);
+
+      const input = getByTestId('DesignSystem-Input');
+      expect(input).not.toHaveClass('mr-4');
+    });
+  });
+
+  describe('Icon Size Calculation Logic', () => {
```

Comment:
Please write proper description for all the test cases. Also specify the **"Input Component"** in the description to clearly identify that this test case belongs to which component.